app/scripts/views/settings/recovery_codes.js

```diff
@@ -27,6 +27,13 @@ const View = FormView.extend({
   },
 
   _returnToTwoStepAuthentication() {
+    // If this view was navigated from `sign_in_recovery_code`, that
+    // means that the user came here to generate new recovery codes,
+    // continue the sign-in process after generating.
+    if (this.model.get('previousViewName') === 'sign_in_recovery_code') {
```

Comment:
Should `sign_in_recovery_code` come from `lib/constants`? What about `previousViewname`?

(also `afterCompleteSignInWithCode` and `settings/two_step_authentication` for that matter)

Reply:
Hmm I opted not too mainly because other views/brokers methods are called this way (not really a strong argument). If I made these a constant, I would want to update the others as well.